### DISCUSSION TITLE
How to create a blank or empty message object

### DISCUSSION THREAD

**Discussion by ChuckCitrano** [nextgenhealthcare/connect#6305] [General]
Hello,

Struggling with something that seems it should be simple, but proving to be elusive.  In the code below, the msgPayload is the msg object being passed into the function.   I'm trying to create an empty version of it so that I can have a blank message object of same type as incoming msg object.  What I'm struggling with is that the only way I can get a empty version is to make a copy of msgPayload via the .copy() method and then delete all children of it.  This works just fine on reasonable size files, but can be really slow when working with a larger file.  Are there simple ways of creating the empty object so that I can use methods like .copy(), appendChild(), etc ...

Thanks in advance.

Chuck

````
function xti_parseclaimsfile(msgPayload) {
	var claimString = '';
	var segmentList = '';
	var claimFinished = false;
	var currentLoop = ''
	var claimsList = [];
	
	globalChannelMap.put('CLMProcessed',  claimsList.length);
	globalChannelMap.put('CLMTotal',  msgPayload['CLM'].length());

	emptyMsg = msgPayload.copy();

	cntrr = 0;
	globalChannelMap.put('SegDeleted', cntrr);
	for each (seg in emptyMsg.children()) {
		delete emptyMsg.children()[seg.childIndex()];
		globalChannelMap.put('SegDeleted',  cntrr++);
	}

	fileHeaderMsg = emptyMsg.copy();
	stLoopMsg = emptyMsg.copy();
	hl20LoopMsg = emptyMsg.copy();
	hl22LoopMsg = emptyMsg.copy();
	clm2300LoopMsg = emptyMsg.copy();

        .... just more parsing logic

`
```

**Comment by tonygermano** [comment]
I think you're going to have to more clearly define what you consider an empty message. Unless your data type is set to Raw or JSON (and I think you're using edi/x12 based on the variables names) then `msg` is an e4x xml object. The way you create segments and fields is going to be basically the same as if you were working with hl7 (also xml-based in mirth,) which you can probably find way more examples of in other posts.

**Comment by ChuckCitrano** [comment]
Tony,

Thanks for offering advice.  I was able to accomplish what I needed with a very simple ...

```
	fileHeaderMsg = new XML('<X12Transaction/>');
	stLoopMsg = new XML('<X12Transaction/>');
	hl20LoopMsg = new XML('<X12Transaction/>');
	hl22LoopMsg = new XML('<X12Transaction/>');
	clm2300LoopMsg = new XML('<X12Transaction/>');
```

My code parses 837 files and breaks them down to individual claims.  I do that by looping through all of the segments and when I reach an end of claim condition, I reinitialize the different parts to what I was calling an "empty message".  After appending to each of the objects above, I would ultimately take the accumulated XML and do the following to render 837 strings ...

```
			claimStringFHpart = SerializerFactory.getSerializer('EDI/X12').fromXML(fileHeaderMsg);
			claimStringSTpart = SerializerFactory.getSerializer('EDI/X12').fromXML(stLoopMsg);
			claimStringHL20part = SerializerFactory.getSerializer('EDI/X12').fromXML(hl20LoopMsg);
			claimStringHL22part = `SerializerFactory.getSerializer('EDI/X12').fromXML(hl22LoopMsg);`
			claimString2300part = SerializerFactory.getSerializer('EDI/X12').fromXML(clm2300LoopMsg);
			claimString = claimStringFHpart + claimStringSTpart + claimStringHL20part + claimStringHL22part + claimString2300part;
			claimsList.push(claimString);
```

Sort of my own crazy logic ... but has held up over time and just needed to figure out a way to create the "empty message" because the payload sizes kept growing and my old method of copying the messages and deleting all the segments was taking too long.

Cheers.
Chuck
